src/aap_eda/api/views/decision_environment.py

```diff
@@ -117,3 +109,26 @@ def retrieve(self, request, pk):
                 decision_environment.data
             ).data
         )
+
+    @extend_schema(
+        description="Delete a decision environment by id",
+        responses=status.HTTP_204_NO_CONTENT,
+    )
+    def destroy(self, request, *args, **kwargs):
+        instance = self.get_object()
+        force_delete = request.query_params.get("force", False)
+
+        activations = models.Activation.objects.filter(
+            decision_environment_id=instance.id
+        ).values_list("name", flat=True)
+
+        if activations and not force_delete:
+            raise api_exc.Conflict(
+                "Decision Environment cannot be deleted; "
+                "it is being used by the following Activations: "
+                f"{list(activations)}. If you want to force delete, "
```

Comment:
We should not put here the activations. might be many. It is simpler just to say that are being used by activations. The api rest is considered a lower level interface.

Reply:
removed.